lib/core/src/sdk.rs

```diff
@@ -4568,6 +4587,161 @@ impl LiquidSdk {
     pub fn init_logging(log_dir: &str, app_logger: Option<Box<dyn log::Log>>) -> Result<()> {
         crate::logger::init_logging(log_dir, app_logger)
     }
+
+    /// Prepares to swap funds from L-BTC to a [TradeableAsset]
+    ///
+    /// # Arguments
+    ///
+    /// * `req` - the [PrepareAssetSwapRequest] containing:
+    ///     * `asset` - The asset we are swapping L-BTC for
+    ///     * `payer_amount_sat` - The amount of L-BTC (in satoshis) we are swapping
+    ///
+    /// # Returns
+    ///
+    /// A [PrepareAssetSwapResponse] containing:
+    ///     * `asset_swap` - the [AssetSwap] details of the ongoing swap, including the exchange
+    ///     rate, fees and the final amount you will receive
+    pub async fn prepare_asset_swap(
+        &self,
+        req: &PrepareAssetSwapRequest,
+    ) -> Result<PrepareAssetSwapResponse, PaymentError> {
+        self.sideswap_service.clone().start().await?;
+
+        let asset_id = req.asset.try_to_asset_id(self.config.network)?;
+
+        ensure_sdk!(
+            self.persister
+                .get_asset_metadata(&asset_id.to_string())
+                .is_ok_and(|a| a.is_some()),
+            PaymentError::asset_error("Asset not found")
+        );
+
+        let asset_swap = self
+            .sideswap_service
+            .subscribe_price_stream(asset_id, req.payer_amount_sat)
+            .await?;
+        if self.get_info().await?.wallet_info.balance_sat
+            < req.payer_amount_sat + asset_swap.fees_sat
+        {
+            self.sideswap_service.unsubscribe_price_stream().await?;
+            return Err(PaymentError::InsufficientFunds);
+        }
+
+        Ok(PrepareAssetSwapResponse { asset_swap })
+    }
+
+    /// Executes a previously prepared asset swap
+    ///
+    /// # Arguments
+    ///
+    /// * `req` - A [ExecuteAssetSwapRequest], containing:
+    ///     * `prepare_response` - the [PrepareAssetSwapResponse] returned by
+    ///       [LiquidSdk::prepare_asset_swap]
+    ///
+    /// # Returns
+    ///
+    /// A [ExecuteAssetSwapResponse] containing:
+    ///     * `payment` - the onchain [Payment] details of the swap, including the Liquid txid and
+    ///     amount
+    ///
+    /// # Errors
+    ///
+    /// * [PaymentError::InvalidOrExpiredFees] - returned if the swap fees have changed above the
+    ///   previously accepted amount
+    /// * [PaymentError::InsufficientFunds] - returned if you do not have enough funds to execute
+    ///   the payment
+    pub async fn execute_asset_swap(
```

Comment:
Like I already mentioned in the other PR, I think here the name could better fit the existing naming scheme.

And the prepare one: `prepare_swap_asset`

```suggestion
    pub async fn swap_asset(
```